### PR TITLE
Add DHT query subcommand to get closest peers

### PR DESCRIPTION
**Pull request by firmianavan**
Simply wrap the same subcommand in IPFS as mentioned in #2177
Please check if there is anything that may need changing.

### CODE REVIEW

File: commands/dht.go [hunk 1]
```diff
@@ -27,9 +28,72 @@ var dhtCmd = &cmds.Command{
...
 	},
 }
+var queryDhtCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline:          "Find the closest Peer IDs to a given Peer ID by querying the DHT.",
+		ShortDescription: "Outputs a list of newline-delimited Peer IDs.",
+	},
+
+	Arguments: []cmdkit.Argument{
+		cmdkit.StringArg("peerID", true, true, "The peerID to run the query against."),
```
Comment (ZenGround0) [inline commands/dht.go]: Please set the third argument `false`.  go-ipfs defaults all of these to variadic even though the command body only every uses the first request argument which seems inconsistent.

File: filnet/router.go [hunk 2]
```diff
@@ -39,3 +42,12 @@ func NewRouter(r routing.IpfsRouting) *Router {
 func (r *Router) FindProvidersAsync(ctx context.Context, key cid.Cid, count int) <-chan pstore.PeerInfo {
 	return r.routing.FindProvidersAsync(ctx, key, count)
 }
+
+// GetClosestPeers returns peers of the K closest peers to the given key
```
Comment (ZenGround0) [inline filnet/router.go]: To help make it clear what K is please modify to something like "GetClosestPeers returns a channel of the K closest peers to the given key where K is the 'K Bucket' parameter of the Kademlia DHT protocol.

Also please be sure to put a fullstop at the end of a comment like this that is a complete sentence as some people here are a little OCD about that 😀.

File: commands/dht.go [hunk 3]
```diff
@@ -27,9 +28,72 @@ var dhtCmd = &cmds.Command{
 	},
 
 	Subcommands: map[string]*cmds.Command{
+		"query":     queryDhtCmd,
```
Comment (phritz) [inline commands/dht.go]: we generally order by alpha

File: plumbing/api.go [hunk 4]
```diff
@@ -187,6 +187,11 @@ func (api *API) NetworkFindProvidersAsync(ctx context.Context, key cid.Cid, coun
 	return api.network.FindProvidersAsync(ctx, key, count)
 }
 
+// NetworkGetClosestPeers issues a getClosestPeers query to the filecoin network.
```
Comment (phritz) [inline plumbing/api.go]: @rosalinekarr @anorth once we kill the node api we might want to make a sweep of plumbing and move anything that is not of general utility out to a direct dependency of the command that uses it, eg this and some of the other network calls.

**Comment by firmianavan** [comment]
> Thanks! my only question is whether it is possible to write a test for this...? generally if we have some functionality, esp with a nontrivial implementation, we want it tested. since this is more for debugging it is less critical, esp since it is copy pasted from ipfs, but still worth seeing if we could write one easily. if so, we should.

Sorry for late response, I am working on the dht test

**Comment by anorth** [comment]
@ZenGround0 could you please shepherd this PR to completion?

**Comment by ZenGround0** [comment]
Hey @firmianavan this is looking good.  Have you started writing a test for this?  If not I will go ahead and merge this and create a follow on issue for testing.

**Comment by firmianavan** [comment]
> Hey @firmianavan this is looking good. Have you started writing a test for this? If not I will go ahead and merge this and create a follow on issue for testing.

I have not finish yet, please do merge and create a follow issue

**Comment by ZenGround0** [comment]
@firmianavan would you mind rebasing on master and then pinging me?

**Comment by firmianavan** [comment]
@ZenGround0 It was done now

**Pull request merged**